src/map/skill.c

```diff
@@ -3514,12 +3518,12 @@ static int skill_attack(int attack_type, struct block_list *src, struct block_li
 	map->freeblock_lock();
 
 	// Plagiarism and Reproduce Code Block [KeiKun]
-	if (damage > 0 && dmg.flag&BF_SKILL && tsd
+	if (bl->type == BL_PC && damage > 0 && dmg.flag&BF_SKILL && tsd
```

Comment:
Talking only about Plagiarism:
Um so you checked that in pre-renewal and renewal one can't copy skills from monsters, even if they're normal playerskills?

This seems wrong to me, at least for pre-renewal.

Reply:
after weeks, i was able to test this under renewal. `bl->type` here is the target which is the player. you can still copy skills from monsters.